src/modules/site/base/views/user.py

```diff
@@ -41,6 +45,39 @@ def user_register():
   return render_template('user/register.html', **locals())
 
 
+@user_bp.route("/recover", methods=["GET", "POST"])
+def user_recover():
+  """ Enter an email address for a user to send a password reset link """
+  title = "Recover User"
+  disable_parent_breadcrumb = True
+  form = RecoverUserForm(request.form)
+
+  if request.method == 'POST' and form.validate():
+    email = request.form.get("email")
+    users = get_user_by_email(email)
+
+    if users and users[0]:
+      user = users[0]
+      token = create_one_time_token(id=user.get('username'), destination=url_for('user.user_reset_password'))
+      password_reset_link = url_for('auth.token', token=token, _external=True)
+      try:
+        send_email({
+          "from": "no-reply@elegansvariation.org",
+          "to": [ email ],
+          "subject": "CeNDR Password Reset",
+          "text": PASSWORD_RESET_EMAIL_TEMPLATE.format(email=email, password_reset_link=password_reset_link)
+        })
+        logger.info(f"Sent password reset email: {email}, link: {password_reset_link}")
+      except Exception as err:
+        logger.error(f"Failed to send email: {err}")
+    
+    # Flash info message even if email fails in order to protect from attempts at finding valid emails
+    flash(f"Please check your email ({email}) for a password reset link.", 'info')

```

Comment:
If an exception was caught, on the email sending, we should not notify the users to expect an email. Consider a slightly different workflow that allows for all the code paths to yield the best 'hints' for the end-user.